creme/tree/_nodes/base.py

```diff
@@ -464,8 +452,8 @@ def get_best_split_suggestions(self, criterion, tree):
 
         """
         best_suggestions = []
-        pre_split_dist = self._stats
-        if not tree.no_preprune:
+        pre_split_dist = self.stats
+        if tree.merit_preprune:
```

Comment:
Nice change, it is easier to understand.

Reply:
:D